pallets/proofs-dealer/src/lib.rs

```diff
@@ -44,17 +44,25 @@ pub mod pallet {
         /// The Providers pallet.
         /// To check if whoever submits a proof is a registered Provider.
         type ProvidersPallet: ProvidersInterface<
-            AccountId = Self::AccountId,
-            MerkleHash = Self::MerkleTrieHash,
-            Balance = Self::NativeBalance,
-        >;
+                AccountId = Self::AccountId,
+                MerkleHash = Self::MerkleTrieHash,
+                Balance = Self::NativeBalance,
+            > + shp_traits::MutateProvidersInterface<
```

Comment:
Not really sure how you want to split this then. In the end you need a function to change the root of bucket or change the root of a bsp. The Mutate trait actually has no mention of a bsp, except for the function `change_bsp_root`, which I can rename to `change_provider_global_root`, to infer that that the provider should have some global root associated to them. And keep the `change_bucket_root` separate.

Reply:
If you attempt to change a root of an MSP by calling `change_provider_global_root` it will fail with `NoGlobalRootForProvider`. Meaning it is not a BSP in our case.